File: src/#104/solution.go
package solution

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	m := 1
	m = max(maxDepth(root.Left)+1, maxDepth(root.Right)+1)
	return m
}
func max(a, b int) int {
	if a > b {
		return a
	}

	return b
}

File: src/#104/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var maxDepth = function (root) {
  if (root === null) {
    return 0
  }

  let m = 1
  m = Math.max(maxDepth(root.left) + 1, maxDepth(root.right) + 1)
  return m
}
